Mathlib/CategoryTheory/Limits/Shapes/Grothendieck.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2024 Jakob von Raumer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob von Raumer
+-/
+import Mathlib.CategoryTheory.Grothendieck
+import Mathlib.CategoryTheory.Limits.HasLimits
+
+/-!
+# (Co)limits on the (strict) Grothendieck Construction
+
+* Shows that if a functor `G : Grothendieck F ⥤ H`, with `F : C ⥤ Cat`, has a colimit, and every
+  fiber of `G` has a colimit, then so does the fiberwise colimit functor `C ⥤ H`.
+* Vice versa, if a each fiber of `G` has a colimit and the fiberwise colimit functor has a colimit,
+  then `G` has a colimit.
+* Shows that colimits of functors on the Grothendieck construction are colimits of
+  "fibered colimits", i.e. of applying the colimit to each fiber of the functor.
+* Derives `HasColimitsOfShape (Grothendieck F) H` with `F : C ⥤ Cat` from the presence of colimits
+  on each fiber shape `F.obj X` and on the base category `C`.
+
+-/
+
+universe v₁ v₂ v₃ u₁ u₂ u₃
+
+namespace CategoryTheory
+
+variable {C : Type u₁} [Category.{v₁} C]
+variable {F : C ⥤ Cat}
+variable {H : Type u₂} [Category.{v₂} H]
+variable (G : Grothendieck F ⥤ H)
+
+lemma Grothendieck.coherence {X Y : Grothendieck F} (hF : X = Y) :
+    eqToHom hF = { base := eqToHom (by subst hF; rfl), fiber := eqToHom (by subst hF; simp) } := by
+  subst hF
+  rfl
+
+namespace Limits
+
+lemma colimit.ι_eq_eqToHom_comp_ι (F : C ⥤ H) [HasColimit F] {c c' : C} (hc : c = c') :
+    colimit.ι F c = eqToHom (by subst hc; rfl) ≫ colimit.ι F c' := by
+  subst hc
+  simp
```

Comment:
`Grothendieck.coherence` and `colimit.ι_eq_eqToHom_comp_ι` should probably be placed in different files. `#find_home!` suggests `Mathlib.CategoryTheory.Grothendieck` for the former and `Mathlib.CategoryTheory.Limits.HasLimits` for the latter.

I also wonder whether the direction of `colimit.ι_eq_eqToHom_comp_ι` should be reversed (and it should probably be tagged `@[reassoc]`).

Reply:
Ah yes, the current state is that I didn't really know whether they deserved to be more than private auxiliary lemmas. Are they really?